src/tests/ftest/control/dmg_pool_query_ranks.py

```diff
@@ -0,0 +1,161 @@
+
+        # Init the pool
+        self.add_pool(connect=False)
+
+    def test_pool_query_ranks_basic(self):
+        """Test the state of ranks with dmg pool query
+
+        Test Description:
+            Create a pool with some engines and check they are all enabled.
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=dmg,pool_query,pool_query_ranks
+        :avocado: tags=pool_query_ranks_basic
+        """
+        self.log.info("Basic tests of pool query with ranks state")
+
+        self.log.debug("Checking without ranks state information")
+        data = self.dmg.pool_query(self.pool.identifier)
+        self.assertIsNone(data['response']['enabled_ranks'],
+                "Invalid enabled_ranks field: want=None, "
+                "got={}".format(data['response']['enabled_ranks']))
+        self.assertIsNone(data['response']['disabled_ranks'],
+                "Invalid disabled_ranks field: want=None, "
+                "got={}".format(data['response']['disabled_ranks']))
+
+        self.log.debug("Checking enabled ranks state information")
+        data = self.dmg.pool_query(self.pool.identifier, show_enabled=True)
+        self.assertListEqual(data['response']['enabled_ranks'], [0, 1, 2],
+                "Invalid enabled_ranks field: want=None, "
+                "got={}".format(data['response']['enabled_ranks']))
+        self.assertIsNone(data['response']['disabled_ranks'],
+                "Invalid disabled_ranks field: want=None, "
+                "got={}".format(data['response']['disabled_ranks']))
+
+        self.log.debug("Checking disabled ranks state information")
+        data = self.dmg.pool_query(self.pool.identifier, show_disabled=True)
+        self.assertIsNone(data['response']['enabled_ranks'],
+                "Invalid enabled_ranks field: want=None, "
+                "got={}".format(data['response']['enabled_ranks']))
+        self.assertListEqual(data['response']['disabled_ranks'], [],
+                "Invalid disabled_ranks field: want=None, "
+                "got={}".format(data['response']['disabled_ranks']))
+
+    def test_pool_query_ranks_error(self):
+        """Test that ranks state option are mutually exclusive
+
+        Test Description:
+            Check that options '--show-enabled' and '--show-disabled" are muttually exclusive.
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=dmg,pool_query,pool_query_ranks
+        :avocado: tags=pool_query_ranks_error
+        """
+        self.log.info("Tests of pool query with incompatible options")
+
+        # Disable raising an exception if the dmg command fails
+        self.dmg.exit_status_exception = False
+        try:
+            data = self.dmg.pool_query(self.pool.identifier, show_enabled=True, show_disabled=True)
+            self.assertIsNotNone(data["error"], "Invalid error field: want={}, "
+                    "got=None".format(data['error']))
```

Comment:
Here `data["error"]` is what we got, rather than what we want.

Reply:
- [ ] Fix invalid assertion comment